spirv_cross/src/spirv.rs

```diff
@@ -188,6 +202,23 @@ where
             .get_cleansed_entry_point_name(entry_point_name)
     }
 
+    /// Gets all specialization constants.
+    pub fn get_specialization_constants(&self) -> Result<Vec<SpecializationConstant>, ErrorCode> {
+        self.compiler.get_specialization_constants()
+    }
+
+    /// Get a mutable reference to a scalar constant.
+    ///
+    /// Can be used to override specialization constants by changing the `data` in the returned
+    /// constant reference.
+    ///
+    /// # Unsafety
+    ///
+    /// The returned `ScalarConstant` holds a **mutable** reference to an AST internal data value.
```

Comment:
Why not just have a `set_scalar_constant` and avoid this reference?

Reply:
Good point! I assumed at an earlier state that I need to set the constants and reset them afterwards (so read/write) but now I don't need it. `set_scalar_constants` sounds good, I'll change it.

Reply:
Sounds good. We could keep the `get` to be consistent with other API functions, but return the value instead of the mutable reference.

Reply:
Ah I see you already rebased, don't worry about `get`, we can always add it later.